### PR TITLE
feat(atst): Seperate react into a different build target

### PR DESCRIPTION
**Pull request by roninjin10**
Anybody using atst is forced to install react even if they aren't using react.

- seperate react/wagmi into it's seperate build target
- now users import react

import { useReadMyContract } from '@eth-optimism/atst/react'

instead of 

import { useReadMyContract } from '@eth-optimism/atst'

Everything else is unchanged except the path to the export

We build the react entrypoint seperately in tsupconfig and then we add the export paths to package.json

### CODE REVIEW

**Comment by roninjin10** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #5045** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/5045" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/5045?utm_source=stack-comment).

File: packages/atst/src/index.ts [hunk 1]
```diff
@@ -28,5 +28,3 @@ export type { AttestationCreatedEvent } from './types/AttestationCreatedEvent'
 export type { AttestationReadParams } from './types/AttestationReadParams'
 export type { DataTypeOption } from './types/DataTypeOption'
 export type { WagmiBytes } from './types/WagmiBytes'
-// react
-export * from './react'
```
Comment (roninjin10) [inline packages/atst/src/index.ts:32]: this makes it so if you aren't using react you don't need to install wagmi or react

File: packages/atst/src/types/ParseBytesReturn.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import { BigNumber } from 'ethers'
-import { Address } from 'wagmi'
+import type { Address } from '@wagmi/core'
```
Comment (roninjin10) [inline packages/atst/src/types/ParseBytesReturn.ts:2]: changing these imports to make sure we don't accidently import wagmi

File: packages/atst/tsup.config.ts [hunk 3]
```diff
@@ -10,7 +10,7 @@ export default defineConfig({
    *
    * @see https://tsup.egoist.dev/#building-cli-app
    */
-  entry: ['src/index.ts', 'src/cli.ts'],
+  entry: ['src/index.ts', 'src/cli.ts', 'src/react.ts'],
```
Comment (roninjin10) [inline packages/atst/tsup.config.ts:13]: build react seperately

File: packages/atst/package.json [hunk 4]
```diff
@@ -6,6 +6,23 @@
...
+      "types": "./src/cli.ts",
+      "main": "./dist/cli.js",
+      "module": "./dist/cli.cjs"
+    },
+    "./react": {
+      "types": "./src/react.ts",
+      "main": "./dist/react.js",
+      "module": "./dist/react.cjs"
+    }
+  },
```
Comment (roninjin10) [inline packages/atst/package.json:20]: this makes it so the imports work like this

import { readAttestation } from '@eth-optimism/atst'
import { readAttestation } from '@eth-optimism/react'

**Review by jakim929 — APPROVED** [review]
+100 Love the separation between vanilla js and react.

**Comment by roninjin10** [comment]
Verified this worked in a canary release in https://github.com/ethereum-optimism/optimism-starter

**Pull request merged**